lib/max_subarray.py

```diff
@@ -9,4 +9,24 @@ def max_sub_array(nums):
         return 0
     if len(nums) == 0:
         return 0
-    pass
+    
+    max_so_far = nums[0]
+    max_ending = 0
+
+    for i in range(len(nums)):
+
+        max_ending = max_ending + nums[i]
+
+        if max_so_far <= max_ending:
+            max_so_far = max_ending
+
+        if max_ending < 0:
+            max_ending = 0
+
+        # Moved before lines 23-24 because arr with all negative values
+        # was resetting max_ending to 0 prematurely
+        
+        # if max_so_far <= max_ending:
+        #     max_so_far = max_ending
+    
+    return max_so_far
```

Comment:
👍  Nice work, you can also use the `max` function to make the code a bit simpler. 

time/space complexity?